internal/pkg/cli/deploy/deploy.go

```diff
@@ -429,25 +439,56 @@ func (d *workloadDeployer) UploadArtifacts() (*UploadArtifactsOutput, error) {
 	}, nil
 }
 
+// GenerateCloudFormationTemplateInput is the input of GenerateCloudFormationTemplate.
+type GenerateCloudFormationTemplateInput struct {
+	StackConfiguration
+}
+
+// GenerateCloudFormationTemplateOutput is the output of GenerateCloudFormationTemplate.
+type GenerateCloudFormationTemplateOutput struct {
+	Template   string
+	Parameters string
+}
+
+// GenerateCloudFormationTemplate genrates a CloudFormation template and parameters for a workload.
+func (d *lbSvcDeployer) GenerateCloudFormationTemplate(in *GenerateCloudFormationTemplateInput) (
+	*GenerateCloudFormationTemplateOutput, error) {
```

Comment:
What do you think of ⬇️ ?
```
func (d *workloadDeployer) GenerateCloudFormationTemplate() ()
```
and 
```
func (d *serviceDeployer) DeployWorkload() ()
```
To extract the common logic shared by all workload/service ?

Reply:
Hmm there aren't many common logics other than two function calls which should be fine 🤔 (since I've already consolidated most common logic in `workloadDeployer.generateCloudFormationTemplate`)  And the main reason is because `stackConfiguration` is defined per workload type so that `workloadDeployer` doesn't have `stackConfiguration` as its method.